tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -1873,6 +1873,51 @@ public:
                                                     convertVector2RVec);
    }
 
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Return the schema of the dataset
+   /// \return the schema of the dataset as string
+   ///
+   /// This convenience function combines the information from GetColumnNames and
+   /// GetColumnType in a table to inspect easily the dataset schema.
+   ///
+   /// This is not an action nor a transformation, just a query to the RDataFrame object.
+   ///
+   /// ### Example usage:
+   /// ~~~{.cpp}
+   /// RDataFrame df(10);
+   /// auto df2 = df.Define("x", "1.f").Define("s", "\"myStr\"");
+   /// // Print schema
+   /// std::cout << df2.GetSchema() << std::endl;
+   /// ~~~
+   ///
+   std::string GetSchema()
+   {
+      const auto columnNames = GetColumnNames();
+      const auto columnTypes = GetColumnTypeNamesList(columnNames);
+
+      // Compute optimal width for the column names
+      const auto n = columnNames.size();
+      auto columnWidth = 0u;
+      for (const auto& name : columnNames) {
+         const auto length = name.length();
+         if (length > columnWidth)
+             columnWidth = length;
+      }
```

Comment:
```cpp
const auto columnWidth = std::max_element(columnNames.begin(), columnNames().end(), [] (const std::string &s) { return s.length(); })->length();
```

or maybe not...damn STL algorithms are unreadable

Reply:
That's why I stick to good old C style coding if it's not magic ;)